components/text-field/script.js

```diff
@@ -0,0 +1,504 @@
+        this.inputElement.addEventListener('blur', this.hideNumberControlBound);
+        this.inputElement.addEventListener('input', this.onNumberInputChangedBound);
+    }
+
+    /**
+     * Filter the input value and replace all characters that are not number or '.'
+     * @param {Event} event
+     */
+    onNumberInputChanged(event) {
+        event.target.value = event.target.value.replace(/[^0-9.]/g, '').replace(/(\..*?)\..*/g, '$1');
+    }
+
+    /**
+     * Will show/hide the placeholder of the input
+     * @param {boolean} visible
+     */
+    togglePlaceholder(visible) {
+        if (!this.placeholder) return;
+
+        if (visible) {
+            this.placeholderElement.classList.remove('guic-hidden');
+        } else {
+            this.placeholderElement.classList.add('guic-hidden');
+        }
+
+    }
+
+    /**
+     * Will clear interval and timeout that are set when the arrow from input number control is pressed
+     */
+    clearMousedownInterval() {
+        document.removeEventListener('mouseup', this.clearMousedownIntervalBound);
+
+        if (this.mousedownTimeout) {
+            clearTimeout(this.mousedownTimeout);
+            this.mousedownTimeout = null;
+        }
+
+        if (this.mousedownInterval) {
+            clearInterval(this.mousedownInterval);
+            this.mousedownInterval = null;
+        }
+    }
+
+    /**
+     * Adds an interval after a small timeout that will constantly bump up or down the number input value until the mouse is released
+     * from the arrow of the input control
+     * @param {Function} stepCallback - Should be some of stepUp or stepDown functions.
+     */
+    addAutoStepInterval(stepCallback) {
+        document.addEventListener('mouseup', this.clearMousedownIntervalBound);
+
+        if (!this.mousedownTimeout) {
+            this.mousedownTimeout = setTimeout(() => {
+                if (!this.mousedownInterval) {
+                    this.mousedownInterval = setInterval(stepCallback.bind(this), 50);
+                }
+            }, 200)
+        }
+    }
+
+    /**
+     * Will set the auto step interval and bump up the input value
+     * @param {Event} event
+     */
+    onStepUp(event) {
+        this.addAutoStepInterval(this.stepUp);
+        this.stepUp();
+        event.preventDefault();
+    }
+
+    /**
+     * Will set the auto step interval and bump down the input value
+     * @param {Event} event
+     */
+    onStepDown(event) {
+        this.addAutoStepInterval(this.stepDown);
+        this.stepDown();
+        event.preventDefault();
+    }
+
+    /**
+     * Will bump up the input value based on the min and max restrinctions
+     */
+    stepUp() {
+        if (this.type !== supportedTextFieldTypes.NUMBER) return;
+
+        const currentValue = this.value === '' ? 0 : parseFloat(this.value);
+
+        if (isNaN(currentValue)) return;
+
+        const nextStep = this.step + currentValue;
```

Comment:
How do we get the step? There's no getter.

Reply:
You don't need a getter. Makin `this.step` is enough to get the step. Note that `initTextFieldAttribute` will propagate attribute name and value to the this context like `this[attributeName] = attributeValue` when the text-field is initialized. `step` will be also set to the `this` context from this method.